### PR TITLE
add math function square

### LINKED ISSUE #3181: provide math function SQUARE in StarRocks
`SQUARE` Definition: https://docs.microsoft.com/en-us/sql/t-sql/functions/square-transact-sql?view=sql-server-ver15

We can refer to the log2 implementation

1. declare function `square` in be/src/exprs/vectorized/math_functions.h
2. implentment function `square` in be/src/exprs/vectorized/math_functions.cpp
3. Register function `square` in gensrc/script/vectorized/vectorized_functions.py

Welcome newcomers to solve this issue
If you have any difficulties you can contact me (stdpain) in slack (https://app.slack.com/client/T02EUPUD8FP/C02FAD0JSSD)

### PR DESCRIPTION
**Pull request by blackstar-baba**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：
Fixes #3181

## Problem Summary(Required) ：
provide math function SQUARE

### CODE REVIEW

**Comment by blackstar-baba** [comment]
@Seaven PTAL.  Happy Chinese New Year !

File: be/src/exprs/vectorized/math_functions.cpp [hunk 1]
```diff
@@ -153,6 +153,13 @@ DEFINE_UNARY_FN_WITH_IMPL(log2Impl, v) {
 
 DEFINE_MATH_UNARY_WITH_OUTPUT_NAN_CHECK_FN(log2, TYPE_DOUBLE, TYPE_DOUBLE);
 
+// square
+DEFINE_UNARY_FN_WITH_IMPL(squareImpl, v) {
+    return (double)std::pow(v, 2.0);
```
Comment (ThunderScar) [inline be/src/exprs/vectorized/math_functions.cpp:158]:
```suggestion
    return v * v;
```

Comment (stdpain) [inline be/src/exprs/vectorized/math_functions.cpp:158]:
```suggestion
    return v * v;
```

Comment (blackstar-baba) [reply be/src/exprs/vectorized/math_functions.cpp:158]: OK，i will modify to v*v to improve performance

**Comment by blackstar-baba** [comment]
I see now that there are 2 commits in the PR, do I need to merge the commit?

**Review by stdpain — APPROVED** [review]

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**